### ISSUE TITLE
No loss while training - Wrong dataset structure?

### ISSUE THREAD

**Issue by KatieBelli** [ZFTurbo/Music-Source-Separation-Training#126]
I use dataset type 1 (even though I'd like to use type 4, but I don't know how to change it).

Somehow there are no changes in loss while training:

![Image](https://github.com/user-attachments/assets/47b01c7d-a59d-48e8-a488-7772f3a24e63)

My structure for training is:

```
training (folder which includes all song subfolders)

song name 1 (folder which includes the 2 flac files)

instruments.flac
vocals.flac
```

My structure for training is:

```
validation (folder which includes all song subfolders)

song name 2 (folder which includes the 3 flac files)

instruments.flac
vocals.flac
mixture.flac
```

I named the "instruments" and "vocals" in the config file as "instruments:".
Batch size is 2. Gradient accumulation steps is 3.

**Comment by ZFTurbo** [comment]
Currently I can't say what's wrong. Training must be normal. May be we need to look into config file. Also check that all training files are correct has enough length and not silent.

**Comment by KatieBelli** [comment]
> Currently I can't say what's wrong. Training must be normal. May be we need to look into config file. Also check that all training files are correct has enough length and not silent.

My config file for training a pretrained MDX23C model:

```
audio:
  chunk_size: 130560
  dim_f: 1024
  dim_t: 256
  hop_length: 512
  n_fft: 2048
  num_channels: 2
  sample_rate: 44100
  min_mean_abs: 0.001

model:
  act: gelu
  bottleneck_factor: 4
  growth: 128
  norm: InstanceNorm
  num_blocks_per_scale: 2
  num_channels: 128
  num_scales: 5
  num_subbands: 4
  scale:
  - 2
  - 2

training:
  batch_size: 2
  gradient_accumulation_steps: 3
  grad_clip: 0
  instruments:
  - instruments
  - vocals
  lr: 0.001
  patience: 15
  reduce_factor: 0.95
  target_instrument: Similarity
  num_epochs: 280
  num_steps: 2235
  q: 0.95
  coarse_loss_clip: true
  ema_momentum: 0.999
  optimizer: prodigy
  other_fix: false # it's needed for checking on multisong dataset if other is actually instrumental
  use_amp: true # enable or disable usage of mixed precision (float16) - usually it must be true

inference:
  batch_size: 8
  dim_t: 256
  num_overlap: 8
```

All files are correctly synced and if there is silence then it's barely more than 1 or 2 seconds. The shortest song length in the training dataset is one and a half minutes.

**Comment by ZFTurbo** [comment]
Set ` target_instrument: null`

**Comment by KatieBelli** [comment]
Maybe it's because I only ha

> Set ` target_instrument: null`. Also `lr` for `prodigy` optimizer must be 1.0

Great, thank you! It works now. May I ask one more question: 
The loss and avg_loss, how do I know that the training is going well? Should those losses get bigger or smaller when training is going well?

**Comment by KatieBelli** [comment]
I just tested it: Is this normal after training one epoch?

```
Training loss: 0.00019843858076433053
0it [00:00, ?it/s]
Num overlap: 8
Instr Similarity sdr: nan (Std: nan)
Instr Difference sdr: nan (Std: nan)
Metric avg sdr        : nan
Elapsed time: 0.01 sec
Train epoch: 1 Learning rate: 1.0
```

**Comment by ZFTurbo** [comment]
You have zero files in validation. If you have `flac` files there, then use `--extension flac`.

Loss values must decrease. Your metrics on validation like SDR must grow.

**Comment by ZFTurbo** [comment]
Try to add in config:

```
inference:
  extension: flac
```

**Comment by KatieBelli** [comment]
I had a typo in the path for the validation dataset, that was the reason for my previous comment. 
I will try to train with flac files the next time, with your suggested extension in the config file. 

Currently, it trains with wav, just to be sure everything is working well and then I will try it with flac. 
Everything works fine now. 

Thank you for helping me quickly.
